repos/fdbt-site/src/pages/api/editPeriodDuration.ts

```diff
@@ -0,0 +1,76 @@
+import { ErrorInfo, NextApiRequestWithSession } from '../../interfaces';
+import { NextApiResponse } from 'next';
+import { getSessionAttribute, updateSessionAttribute } from '../../../src/utils/sessions';
+import {
+    EDIT_PERIOD_DURATION_ERROR,
+    MATCHING_JSON_ATTRIBUTE,
+    MATCHING_JSON_META_DATA_ATTRIBUTE,
+} from '../../../src/constants/attributes';
+import { redirectTo, redirectToError } from '../../../src/utils/apiUtils';
+import { putUserDataInProductsBucketWithFilePath } from '../../utils/apiUtils/userData';
+import {
+    checkDurationIsValid,
+    isValidInputDuration,
+    removeExcessWhiteSpace,
+} from '../../../src/utils/apiUtils/validator';
+
+const validateDuration = (productDuration: string, productDurationUnit: string): ErrorInfo[] => {
+    const trimmedDuration = removeExcessWhiteSpace(productDuration);
+    const errors = [];
+    const productDurationError = checkDurationIsValid(trimmedDuration);
+    const productDurationUnitError = !isValidInputDuration(productDurationUnit, false)
+        ? 'Choose an option from the dropdown'
+        : '';
+
+    if (productDurationError) {
+        errors.push({ id: 'product-duration', errorMessage: productDurationError });
+    }
+    if (productDurationUnitError) {
+        errors.push({ id: 'product-duration-unit', errorMessage: productDurationUnitError });
+    }
+    return errors;
+};
+export default async (req: NextApiRequestWithSession, res: NextApiResponse): Promise<void> => {
+    try {
+        const ticket = getSessionAttribute(req, MATCHING_JSON_ATTRIBUTE);
+        const matchingJsonMetaData = getSessionAttribute(req, MATCHING_JSON_META_DATA_ATTRIBUTE);
+        if (!ticket || !matchingJsonMetaData) {
+            throw new Error('Ticket information cannot be undefined');
+        }
+        const productDurationValue = req.body['multipleProductDurationInput'];
```

Comment:
Is there a reason we're doing this instead of req.body.multipleProductDurationInput?